src/ServiceControl.Monitoring/Http/Diagrams/DiagramApiModule.cs

```diff
@@ -81,22 +86,43 @@ public MonitoredEndpointsModule(IProvideBreakdownBy<EndpointInstanceId>[] metric
                     digest.Metrics.Add(metric.ReturnName, new MonitoredEndpointMetricDigest{Latest = values.Points.LastOrDefault(), Average = values.Average});
                 }
 
-                foreach (var metric in instanceMetric)
+                var metricDetails = new MonitoredEndpointMetricDetails();
+
+                foreach (var metric in instanceMetrics)
                 {
                     var store = metricByInstanceLookup[metric.StoreType];
-                    var intervals = store.GetIntervals(period, DateTime.UtcNow).ToLookup(k => k.Id.InstanceId);
+                    var intervals = store.GetIntervals(period, DateTime.UtcNow);
```

Comment:
This is the second loop over `instanceMetrics`. Notice, that in both cases we obtain intervals. This means two intervals generation an turning them to different lookups. I see no reason to not keep them together and iterate over metrics only once.